src/main/java/core/basesyntax/supplier/FigureSupplier.java

```diff
@@ -0,0 +1,45 @@
+package core.basesyntax.supplier;
+
+import core.basesyntax.figure.Circle;
+import core.basesyntax.figure.Figure;
+import core.basesyntax.figure.IsoscelesTrapezoid;
+import core.basesyntax.figure.Rectangle;
+import core.basesyntax.figure.RightTriangle;
+import core.basesyntax.figure.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int LIMIT_RANDOM = 15;
+    private static final int NUMBERS_OF_FIGURES = 5;
+    private final Random random;
+    private final ColorSupplier colorSupplier;
+
+    public FigureSupplier() {
+        random = new Random();
+        colorSupplier = new ColorSupplier();
+    }
```

Comment:
Initialize `Random` and `ColorSupplier` directly in field variables or with **constructor with params.**